src/test/java/com/coinbase/api/CoinbaseTest.java

```diff
@@ -1,78 +1,52 @@
-import java.util.Map;
-
-import javax.net.ssl.HttpsURLConnection;
-
-import org.apache.commons.io.output.NullOutputStream;
+import com.coinbase.api.entity.*;
+import com.coinbase.api.exception.CoinbaseException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.joda.money.CurrencyUnit;
 import org.joda.money.Money;
 import org.joda.time.DateTime;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
 import org.powermock.core.classloader.annotations.PowerMockIgnore;
 import org.powermock.core.classloader.annotations.PrepareForTest;
 import org.powermock.modules.junit4.PowerMockRunner;
 
-import com.coinbase.api.entity.Account;
-import com.coinbase.api.entity.AccountChange;
-import com.coinbase.api.entity.AccountChangesResponse;
-import com.coinbase.api.entity.AccountsResponse;
-import com.coinbase.api.entity.Address;
-import com.coinbase.api.entity.AddressResponse;
-import com.coinbase.api.entity.AddressesResponse;
-import com.coinbase.api.entity.Application;
-import com.coinbase.api.entity.Button;
-import com.coinbase.api.entity.Contact;
-import com.coinbase.api.entity.ContactsResponse;
-import com.coinbase.api.entity.HistoricalPrice;
-import com.coinbase.api.entity.Merchant;
-import com.coinbase.api.entity.Order;
-import com.coinbase.api.entity.OrdersResponse;
-import com.coinbase.api.entity.PaymentMethod;
-import com.coinbase.api.entity.PaymentMethodsResponse;
-import com.coinbase.api.entity.Quote;
-import com.coinbase.api.entity.RecurringPayment;
-import com.coinbase.api.entity.RecurringPaymentsResponse;
-import com.coinbase.api.entity.Report;
-import com.coinbase.api.entity.Token;
-import com.coinbase.api.entity.Transaction;
-import com.coinbase.api.entity.TransactionsResponse;
-import com.coinbase.api.entity.Transfer;
-import com.coinbase.api.entity.TransfersResponse;
-import com.coinbase.api.entity.User;
-import com.coinbase.api.exception.CoinbaseException;
+import javax.net.ssl.HttpsURLConnection;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.powermock.api.mockito.PowerMockito.*;
 
 @RunWith(PowerMockRunner.class)
 @PowerMockIgnore("javax.net.ssl.*")
-@PrepareForTest(CoinbaseImpl.class)
+@PrepareForTest({CoinbaseImpl.class, URL.class})
 public class CoinbaseTest {
 
+    private static final ObjectMapper objectMapper = ObjectMapperProvider.createDefaultMapper();
     private HttpsURLConnection mockConnection;
-    private URL mockURL;
     private Coinbase cb;
+    private ByteArrayOutputStream requestStream;
+    private ArgumentCaptor<String> urlCaptor;
 
     @Before
     public void setUp() throws Exception {
         mockConnection = mock(HttpsURLConnection.class);
-        mockURL = mock(URL.class);
+        URL mockURL = mock(URL.class);
```

Comment:
Use mockito f/w when possible. It has internal mocking of objects so you don't have to write the high level wrappers for yourself.